### PR TITLE
ZCS-5825:Create Alias for zmprov HAB commands

### PR DESCRIPTION
**Pull request by shrikant-prasad**
Added new commands to:
1. add hab group member
2. delete hab group
3. get hab group members
4. remove hab group member

now list of all HAB commands are:

  createHABOrgUnit(chou) {domain} {ouName}
  renameHABOrgUnit(rhou) {domain} {ouName} {newName}
  deleteHABOrgUnit(dhou) {domain} {ouName}
  createHABGroup(chg) {groupName} {ouName} {name@domain} {TRUE|FALSE} [attr1 value1 [attr2 value2...]]
  getHAB(ghab) {habRootGrpId}
  moveHABGroup(mhg) {habRootGrpId} {habParentGrpId} {targetHabParentGrpId}
  addHABGroupMember(ahgm) {name@domain|id} {member@domain}+
  removeHABGroupMember(rhgm) {name@domain|id} {member@domain}
  deleteHABGroup(dhg) {name@domain|id} [true|false]
  modifyHABGroupSeniority(mhgs) {habGrpId} {seniorityIndex} 
  getHABGroupMembers(ghgm) {name@domain|id}

### CODE REVIEW

File: store/src/java/com/zimbra/cs/account/ProvUtil.java [hunk 1]
```diff
@@ -1623,13 +1629,41 @@ private boolean execute(String args[]) throws ServiceException, ArgException, IO
...
+        case GET_HAB_GROUP_MEMBERS:
+            doGetHABGroupMembers(args);
             break;
         default:
             return false;
         }
         return true;
     }
 
+    private void doAddMember(String[] args) throws ServiceException {
```
Comment (desouzas) [inline store/src/java/com/zimbra/cs/account/ProvUtil.java]: Do we need to add arg length check and usage() call for each of these?

File: store/src/java/com/zimbra/cs/account/ProvUtil.java [hunk 2]
```diff
@@ -857,14 +857,24 @@ public void printHelp() {
...
             "{groupName} {ouName} {name@domain} {TRUE|FALSE} [attr1 value1 [attr2 value2...]]", Category.HAB , 3, Integer.MAX_VALUE),
-        GET_HAB("getHab", "ghab",
-            "{habRootGrpId} ", Category.HAB , 1, 1),
-        MODIFY_HAB_GROUP("modifyHabGroup", "mhab",
-            "{habGrpId} {habParentGrpId} {targetHabParentGrpId} ", Category.HAB , 3, 3),
-        MODIFY_HAB_GROUP_SENIORITY("modifyHabGroupSeniority", "mhsi",
-            "{habGrpId} {seniorityIndex} ", Category.HAB, 2, 2);
+        GET_HAB("getHAB", "ghab",
+            "{habRootGrpId}", Category.HAB, 1, 1),
+        MODIFY_HAB_GROUP("modifyHABGroup", "mhg",
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/account/ProvUtil.java]: Wondering if wee should change at least fro the command from modify to move

File: store/src/java/com/zimbra/cs/account/ProvUtil.java [hunk 3]
```diff
@@ -1623,13 +1629,41 @@ private boolean execute(String args[]) throws ServiceException, ArgException, IO
             break;
         case MODIFY_HAB_GROUP_SENIORITY:
             modifyHabGroupSeniority(args);
+        case ADD_HAB_GROUP_MEMBER:
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/account/ProvUtil.java]: missing break;

File: store/src/java/com/zimbra/cs/account/ProvUtil.java [hunk 4]
```diff
@@ -1623,13 +1629,41 @@ private boolean execute(String args[]) throws ServiceException, ArgException, IO
...
         default:
             return false;
         }
         return true;
     }
 
+    private void doAddMember(String[] args) throws ServiceException {
+        String[] members = new String[args.length - 2];
+        System.arraycopy(args, 2, members, 0, args.length - 2);
+        prov.addGroupMembers(lookupGroup(args[1]), members);
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/account/ProvUtil.java]: do we support it both for soap and ldap rov.

File: store/src/java/com/zimbra/cs/account/ProvUtil.java (hunk 1 above)
Comment (shrikant-prasad) [reply store/src/java/com/zimbra/cs/account/ProvUtil.java]: It's already handled. 
zimbra@sprasad:~$ zmprov adlm 
usage:  addDistributionListMember(adlm) {list@domain|id} {member@domain}+
For general help, type : zmprov --help
zimbra@sprasad:~$ zmprov adlm f02ca108-14d5-4f23-a0c4-53d92c8bebdd
usage:  addDistributionListMember(adlm) {list@domain|id} {member@domain}+
For general help, type : zmprov --help

File: store/src/java/com/zimbra/cs/account/ProvUtil.java (hunk 3 above)
Comment (shrikant-prasad) [reply store/src/java/com/zimbra/cs/account/ProvUtil.java]: fixed

File: store/src/java/com/zimbra/cs/account/ProvUtil.java (hunk 4 above)
Comment (shrikant-prasad) [reply store/src/java/com/zimbra/cs/account/ProvUtil.java]: ldap and soap both are supported.

File: store/src/java/com/zimbra/cs/account/ProvUtil.java (hunk 2 above)
Comment (shrikant-prasad) [reply store/src/java/com/zimbra/cs/account/ProvUtil.java]: changed to moveHABGroup

File: store/src/java/com/zimbra/cs/account/ProvUtil.java [hunk 5]
```diff
@@ -857,14 +857,24 @@ public void printHelp() {
...
+        MODIFY_HAB_GROUP("modifyHABGroup", "mhg",
+            "{habRootGrpId} {habParentGrpId} {targetHabParentGrpId}", Category.HAB , 3, 3),
+        ADD_HAB_GROUP_MEMBER("addHABGroupMember", "ahgm", "{name@domain|id} {member@domain}+",
+                Category.HAB, 2, Integer.MAX_VALUE),
+        REMOVE_HAB_GROUP_MEMBER(
+                "removeHABGroupMember", "rhgm", "{name@domain|id} {member@domain}", Category.HAB, 2,
+                Integer.MAX_VALUE),
+        DELETE_HAB_GROUP("deleteHABGroup", "dhg", "{name@domain|id} [true|false]",
+                Category.HAB, 1, 2),
+        MODIFY_HAB_GROUP_SENIORITY("modifyHABGroupSeniority", "mhgsi",
```
[the remark this answers is not in the document]
Comment (shrikant-prasad) [reply store/src/java/com/zimbra/cs/account/ProvUtil.java]: changed to mhgs

**Pull request closed without merging**